src/log/log.c

```diff
@@ -712,10 +714,28 @@ __log_decompress(WT_SESSION_IMPL *session, WT_ITEM *in, WT_ITEM *out)
 	uncompressed_size = logrec->mem_len;
 	WT_RET(__wt_buf_initsize(session, out, uncompressed_size));
 	memcpy(out->mem, in->mem, skip);
-	WT_RET(compressor->decompress(compressor, &session->iface,
+	ret = compressor->decompress(compressor, &session->iface,
 	    (uint8_t *)in->mem + skip, in->size - skip,
 	    (uint8_t *)out->mem + skip,
-	    uncompressed_size - skip, &result_len));
+	    uncompressed_size - skip, &result_len);
+
+	/*
+	 * If our chosen compressor doesn't work, we may have been running with
+	 * a different compressor last time. Cycle through the available ones
+	 * and see if one of them succeeds.
+	 */
+	if (ret != 0)
+		TAILQ_FOREACH(ncomp, &conn->compqh, q) {
+			ret = ncomp->compressor->decompress(compressor,
```

Comment:
Each failed call results in error messages. Do we want to suppress this?

Reply:
> Each failed call results in error messages. Do we want to suppress this?

Yeah, I think so, it's "expected" behavior.

Of course, we'll want an error on our final failure.